bouncer/authentication/authentication.py

```diff
@@ -0,0 +1,41 @@
+from django.contrib.auth.models import User
+from rest_framework import authentication
+from rest_framework import exceptions
+from authentication import models
+
+
+class Authentication(authentication.TokenAuthentication):
+    def authenticate(self, request):
+        auth = authentication.get_authorization_header(request).split()
+
+        if not auth or auth[0].lower() != self.keyword.lower().encode():
+            return None
+
+        if len(auth) == 1:
+            msg = _("Invalid token header. No credentials provided.")
+            raise exceptions.AuthenticationFailed(msg)
+        elif len(auth) > 2:
+            msg = _("Invalid token header. Token string should not contain spaces.")
+            raise exceptions.AuthenticationFailed(msg)
+
+        try:
+            token = auth[1].decode()
+        except UnicodeError:
+            msg = _(
+                "Invalid token header. Token string should not contain invalid characters."
+            )
+            raise exceptions.AuthenticationFailed(msg)
+
+        return self.authenticate_credentials(token)
+
+    def authenticate_credentials(self, key):
+        model = models.AuthToken
+        token = model.objects.select_related("user").get(key=key)
+
+        if not token.user.is_active:
+            raise exceptions.AuthenticationFailed(_("User inactive or deleted."))
+
+        return (token.user, token)
+
+    def authenticate_header(self, request):
```

Comment:
This seems to be exactly the same, as the parent class' (`TokenAuthentication`) method. In that case, it should be completely removed.